amount.go

```diff
@@ -35,6 +35,9 @@ func (a Amount) Float64() float64 { return float64(a.v) / float64(a.c.scale()) }
 
 func (a Amount) Currency() Currency { return a.c }
 
+// Val returns the raw integer value.
+func (a Amount) Val() int64 { return a.v }
+
```

Comment:
lets use same notation as in fpdecimal. https://github.com/nikolaydubina/fpdecimal/blob/main/fpdecimal.go#L45
```suggestion
func (a Amount) Scaled() int64 { return a.v }

```